### ISSUE TITLE
API does not report any local peers on 0.5.0

### ISSUE THREAD

**Issue by r-vdp** [threefoldtech/mycelium#201]
Since 0.5.0, the API no longer reports any local peers:

```
➜ curl --no-progress-meter http://localhost:8989/api/v1/admin/peers | jq 'map(select(.type != "static"))'
[]
```

I have a local network with three Linux nodes, they all have ports udp/9650, udp/9651 and tcp/9651 open. I ran mycelium with `--debug` on one of them but I don't see any log messages saying anything about local peer discovery.

**Comment by LeeSmet** [comment]
There isn't really a (debug) log atm for when a discovery packet is sent it seems. One thing I'm curious about is if the right interface is used to send the multicast beacon. We bind the socket to all IPv6 interfaces and send one beacon to `ff02::cafe` every minute. If you run `ip -6 r get ff02::cafe`, is it using the correct interface in your setup?

**Comment by r-vdp** [comment]
Yeah that seems to be the problem indeed on my laptop, it's selecting the zerotier interface:
```
➜ ip -6 r get ff02::cafe
multicast ff02::cafe from :: dev ztzlggwhus table local proto kernel src fe80::f063:aff:fe3b:4106 metric 256 pref medium
```

However, on my router and a local server (all in the same local network) the routing seems correct but there are still no local peers:
```
# on router
➜ ip -6 r get ff02::cafe
multicast ff02::cafe from :: dev mycelium table local proto kernel src fe80::3188:2717:7ff9:6f53 metric 256 pref medium

➜ curl --no-progress-meter http://localhost:8989/api/v1/admin/peers | jq 'map(select(.type != "static"))'
[]

# on server
➜ ip -6 r get ff02::cafe
multicast ff02::cafe from :: dev mycelium table local proto kernel src fe80::6155:f3d:497a:5b0d metric 256 pref medium
                                                                                                                                                                            
➜ curl --no-progress-meter http://localhost:8989/api/v1/admin/peers | jq 'map(select(.type != "static"))'
[]
```

**Comment by LeeSmet** [comment]
On your router and server, I'm guessing interface "mycelium" is the tun interface of the mycelium process. That won't work, since it sends the discovery packet into the network itself (and that will return an icmp no route to host). What you want is for the OS to send this over the link which is going out. I'm not entirely sure if there is a way to force this from mycelium itself.

All in all, LL will probably need to be expanded to send the multicast discovery packet out on every (IPv6) interface, similarly to how we currently listen on all interfaces.

**Comment by delandtj** [comment]
> 
> All in all, LL will probably need to be expanded to send the multicast discovery packet out on every (IPv6) interface, similarly to how we currently listen on all interfaces.

Except for interfaces that are tun (pointopoint), lo ?

**Comment by LeeSmet** [comment]
Code is updated to send discovery beacons from multiple interfaces instead of relying on the OS to do the proper thing here. Will be included in 0.5.1

**Comment by r-vdp** [comment]
Great, I'll give that a try!
